torchvision/datasets/folder.py

```diff
@@ -72,8 +73,11 @@ def make_dataset(
 
     if extensions is not None:
 
+        if isinstance(extensions, str):
+            extensions = (extensions,)
+
         def is_valid_file(x: str) -> bool:
-            return has_file_allowed_extension(x, cast(Tuple[str, ...], extensions))
+            return has_file_allowed_extension(x, cast(Collection[str], extensions))
```

Comment:
why do we need to cast here? Aren't the type annotations the same?

Reply:
`mypy` has a long standing issue that it doesn't pick up on outer type restriction in local functions. Thus, although logic excludes `None` and `str` here, inside `is_valid_file` `mypy` still thinks `extensions` has type `Optional[Union[str, Collection[str]]]`. Thus, we need to cast to tell it otherwise.

Reply:
Of course....


🥲

Reply:
I've opted to change the `cast` to a `# type: ignore` to indicate that there is nothing "wrong" on our side, but this is actually a `mypy` issue.